File: dato.config.rb
directory "src/pages" do
  dato.pages.each do |page|
    create_post "#{page.slug}.md" do
      attributes = {
        title: page.title,
        subtitle: page.subtitle,
        position: page.position,
        background_image: page.background_image.url(auto: :compress),
        layout: "../layouts/page.astro"
      }

      attributes.reject! do |_, value|
        value.nil? || value.respond_to?(:empty?) && value.empty?
      end

      frontmatter(:yaml, attributes)
      content(page.body.strip + "\n")
    end
  end
end

WEEKDAYS = {
  monday: "Maanantai",
  tuesday: "Tiistai",
  wednesday: "Keskiviikko",
  thursday: "Torstai",
  friday: "Perjantai",
  saturday: "Lauantai",
  sunday: "Sunnuntai"
}

directory "src/content/timetables" do
  data = {
    heading: dato.timetable.heading,
    timetable: WEEKDAYS.map do |key, translation|
      timeslots = dato.timetable.attributes[key].map do |timeslot|
        "#{timeslot.time} #{timeslot.sport}, #{timeslot.location}"
      end

      {
        day: translation,
        times: timeslots
      }
    end
  }

  create_data_file("current.json", :json, data)
end

File: dato.config.js
module.exports = (dato, root, i18n) => {
  root.directory("src/pages", (pagesDir) => {
    dato.pages.forEach((page) => {
      pagesDir.createPost(`${page.slug}.md`, "yaml", {
        frontmatter: {
          title: page.title,
          subtitle: page.subtitle,
          position: page.position,
          background_image: page.backgroundImage.url({ auto: "compress" }),
          layout: "../layouts/page.astro",
        },
        content: page.body.trim() + "\n",
      });
    });
  });

  const WEEKDAYS = {
    monday: "Maanantai",
    tuesday: "Tiistai",
    wednesday: "Keskiviikko",
    thursday: "Torstai",
    friday: "Perjantai",
    saturday: "Lauantai",
    sunday: "Sunnuntai",
  };

  root.directory("src/content/timetables", (timetablesDir) => {
    const data = {
      heading: dato.timetable.heading,
      timetable: Object.entries(WEEKDAYS).map(([key, translation]) => {
        // console.log(key, translation, dato.timetable.attributes[key]);
        const timeslots = dato.timetable.attributes[key].map((timeslot) => {
          return `${timeslot.time} ${timeslot.sport}, ${timeslot.location}`;
        });

        return {
          day: translation,
          times: timeslots,
        };
      }),
    };

    timetablesDir.createDataFile("current.json", "json", data);
  });
};
